crux_db/crux.sh

```diff
@@ -384,29 +379,38 @@ fi
 echo " "
 echo " "
 echo "Part 2.1: Cleaning up blast results"
-echo "For each set of BLAST 1 and 2 results"
-echo "     Merge and De-replicate by NCBI accession version numbers, and convert to fasta format."
-echo "     Then use entrez-qiime to generate a corresponding taxonomy file, and clean the blast output and taxonomy file to eliminate poorly annotated sequences."
 mkdir -p ${ODIR}/${NAME}_db_filtered/${NAME}_fasta_and_taxonomy/
 mkdir -p ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy
 
-# for each BLAST_out fasta folder
-for str in ${ODIR}/${NAME}_BLAST/${NAME}_*_out
-do
-  str1=${str%_BLAST_out}
-  j=${str1#${ODIR}/${NAME}_BLAST/${NAME}_}
-  echo "${str}"
-  echo "${j}"
-  # reduce size blast files by combining and dereplicating by length.
-  python ${DB}/scripts/combine_and_dereplicate_fasta.py -o ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/${NAME}_${j}_blast_out.fasta -a ${str}/fasta/*_out.fasta
-done
+if [ "${SKIPBLAST}" = "FALSE" ]
+then
+  echo "For each set of BLAST 1 and 2 results"
+  echo "     Merge and De-replicate by NCBI accession version numbers, and convert to fasta format."
+  echo "     Then use entrez-qiime to generate a corresponding taxonomy file, and clean the blast output and taxonomy file to eliminate poorly annotated sequences."
+  # for each BLAST_out fasta folder
+  for str in ${ODIR}/${NAME}_BLAST/${NAME}_*_out
+  do
+    str1=${str%_BLAST_out}
+    j=${str1#${ODIR}/${NAME}_BLAST/${NAME}_}
+    echo "${str}"
+    echo "${j}"
+    # reduce size blast files by combining and dereplicating by length.
+    python ${DB}/scripts/combine_and_dereplicate_fasta.py -o ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/${NAME}_${j}_blast_out.fasta -a ${str}/fasta/*_out.fasta
+  done
 
-# for all of of derepliated blast hits, combine and depreplicate by length
-python ${DB}/scripts/combine_and_dereplicate_fasta.py -o ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/${NAME}_.fasta -a ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/*_blast_out.fasta
-rm ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/*_blast_out.fasta
+  # for all of of derepliated blast hits, combine and depreplicate by length
+  python ${DB}/scripts/combine_and_dereplicate_fasta.py -o ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/${NAME}_.fasta -a ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/*_blast_out.fasta
+  rm ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/*_blast_out.fasta
+else
+  echo "Creating fasta format from blast_db"
+  # create reference fasta from blast
+  blastdbcmd -entry all -db ${BLAST_DB} -out ${ODIR}/${NAME}_db_unfiltered/${NAME}_fasta_and_taxonomy/${NAME}_.fasta
```

Comment:
This part unfortunately is wrong, we need to be using bowtie2 to query against the database using the ecopcr output. This copies the whole database to the output, without doing any queries

Reply:
Instead of doing blastdbcmd in the pipeline I think we should assume the user passes in a BLAST_DB in bt2 format already (so maybe rename BLAST_DB var to NT_DB). Then we will have to rewrite the BLAST sections above to be calling `bowtie2` instead of `blastn`.